plugins/web/opentelemetry-plugin-react-load/src/reactLoad.ts

```diff
@@ -151,21 +167,183 @@ export class ReactLoad extends BasePlugin<unknown> {
+        return;
+      }
+
+      return function patchShouldComponentUpdate(
+        this: React.Component,
+        ...args
+      ): boolean {
+        const span = plugin._createSpanWithParent(this, 'shouldComponentUpdate');
+        const apply = original.apply(this, args);
+        if (span) {
+          span.end();
+        }
+        // if shouldComponentUpdate returns false, the component does not get 
+        // updated and no other lifecycle methods get called
+        if(apply === false){
+          const updatingSpan = plugin._getParentSpan(this);
+          if (updatingSpan) {
+            updatingSpan.updateName(AttributeNames.UPDATING_SPAN);
+            updatingSpan.end();
+            plugin._parentSpanMap.delete(this);
+          }
+        }
+        
+        return apply;
+      };
+    };
+  }
+
+
+  /**
+   * Patches the shouldComponentUpdate lifecycle method
+   */
+  private _patchGetSnapshotBeforeUpdate(){
+    return (original: GetSnapshotBeforeUpdateFunction): GetSnapshotBeforeUpdateFunction => {
+      const plugin = this;
+      if (!original) {
+        this._logger.debug(
+          'getSnapshotBeforeUpdate function was undefined, should always be defined when patching.'
+        );
+        return;
+      }
+
+      return function patchGetSnapshotBeforeUpdate(
+        this: React.Component,
+        ...args
+      ): any {
+        const span = plugin._createSpanWithParent(this, 'getSnapshotBeforeUpdate');
+        const apply = original.apply(this, args);
+        if (span) {
+          span.end();
+        }
+        
+        return apply;
+      };
+    };
+  }
+
+  /**
+   * Patches the componentDidUpdate lifecycle method
+   */
+  private _patchComponentDidUpdate(){
+    return (original: ComponentDidUpdateFunction): ComponentDidUpdateFunction => {
+      const plugin = this;
+      if (!original) {
+        this._logger.debug(
+          'componentDidUpdate function was undefined, should always be defined when patching.'
+        );
+        return;
+      }
+
+      return function patchComponentDidUpdate(
+        this: React.Component,
+        ...args
+      ): void {
+        const span = plugin._createSpanWithParent(this, 'componentDidUpdate');
+        const apply = original.apply(this, args);
+        if (span) {
+          span.end();
+        }
+        const updatingSpan = plugin._getParentSpan(this);
+        if (updatingSpan) {
+          updatingSpan.updateName(AttributeNames.UPDATING_SPAN);
+          updatingSpan.end();
+          plugin._parentSpanMap.delete(this);
+        }
+        return apply;
+      };
+    };
+  }
+
   /**
    * implements patch function
    */
   protected patch() {
     this._logger.debug('applying patch to', this.moduleName, this.version);
     this._reactComponents.forEach(prototype => {
+      if (isWrapped(prototype.render)) {
+        shimmer.unwrap(prototype, 'render');
+        this._logger.debug('removing previous patch from method render');
+      }
```

Comment:
IIRC shimmer just remembers if a method is shimmed, but doesn't store an owner of the modification. Is it possible that this check could accidentally remove functionality you don't expect? Such as another plugin, or custom user code if the client code uses shimmer for aspect oriented patterns or some other reason.

The same would apply to the similar patterns below.